### DISCUSSION TITLE
does wire:navigate renders the component or full page ?

### DISCUSSION THREAD

**Discussion by ZeeshanX4** [livewire/livewire#7885] [1. Help] [answered]
Hello, I have admin dashboard with header, sidebar and footer, all livewire 3 components.  I am using default livewire layout (layout.app) and the content section, which is using {{ $slot }}. And I am using livewire routes.  I have a search input in header nav, in which I write something and then if I navigate to, for example, a link using wire:navigate, the input text is gone, which means the whole page is rendered, including the header whereas ideally only the component should be rendered in place of slot. So my question is does the wire:navigate render the component or full page or may be I am not using the right way. Thank you for your help.

**Comment by danie-ramdhani** [comment, marked as answer]
`wire:navigate` is fetching your page then morphing your page. so it is full page render.

if you just want to refresh your component (i.e. `Livewire/Header.php`), from your main component, use `$dispatch`. here's some example:

```php
class Dashboard extends Component
{
    public function render()
    {
        return <<<'HTML'
	    <div>
              <button type="button" x-on:click="$dispatch('refreshHeader')">Refresh Header</button>
	    </div>
	HTML;
    }
}

```

and then add `\Livewire\Attributes\On` your `Header.php` 

```php
#[\Livewire\Attributes\On('refreshHeader')]
class Header extends Component
{
// ...
}

```

**Reply by ZeeshanX4** [reply]
@danie-ramdhani Thank you for your reply and showing an example. 

Yes now I understand it is fetching the new html and then morphing it. The search input in header will be gone with wire:navigate.
